slice-core/src/main/java/com/cognifide/slice/core/internal/monitoring/ModelUsageData.java

```diff
@@ -0,0 +1,80 @@
+/*-
+ * #%L
+ * Slice - Core
+ * %%
+ * Copyright (C) 2012 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.slice.core.internal.monitoring;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.inject.Key;
+
+public class ModelUsageData {
+
+	private ConcurrentHashMap<Class<?>, ModelUsageData> subModels = new ConcurrentHashMap<Class<?>, ModelUsageData>();
+
+	private AtomicLong count = new AtomicLong();
+
+	private AtomicLong totalTime = new AtomicLong();
+
+	public void addTimeMeasurement(Long timeMeasurement) {
+		count.incrementAndGet();
+		totalTime.addAndGet(timeMeasurement);
+	}
+
+	public long getCount() {
+		return count.get();
+	}
+
+	public long getTotalTime() {
+		return TimeUnit.NANOSECONDS.toMillis(totalTime.get());
+	}
+
+	public ConcurrentHashMap<Class<?>, ModelUsageData> getSubModels() {

```

Comment:
Try to avoid using concrete classes in method signatures. `ConcurrentMap` sounds like a better fit here as it allows more flexibility.

Do we even want to expose this collection to the clients of this class? Maybe a copy would be a better idea? The way it's implemented, one can obtain a reference to this internal map and modify it. Not sure this is the desired result.